src/core/util/latent_see.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/util/latent_see.h"
+
+#include <chrono>
+#include <cstdint>
+#include <ratio>
+
+#include "absl/strings/str_cat.h"
+#include "absl/types/optional.h"
+
+#ifdef GRPC_ENABLE_LATENT_SEE
+namespace grpc_core {
+namespace latent_see {
+
+thread_local std::vector<Log::Event> Log::thread_events_;
+thread_local uint64_t Log::thread_id_ = Log::Get().next_thread_id_.fetch_add(1);
+std::atomic<uint64_t> Flow::next_flow_id_{1};
+
+std::string Log::GenerateJson() {
```

Comment:
Would be nice to see a few tests that assert the JSON is parseable with a few different operations.

Reply:
I'm not sure how to do that without either adding a `latent_see` config to CI, or moving some of the json generation out of the `#ifdef` - in which case we run a (small?) risk of including it in builds we don't want to.

Reply:
Yeah, true. The well-tread path here is setting up a new kokoro job, which is an extra ~10 minutes of typing, but it's not great. I wouldn't want to change the ifdef setup. I'll approve without tests, it's not mission critical if this build configuration fails, or if the JSON isn't right.